internal/cloud/backend_plan.go

```diff
@@ -95,7 +87,25 @@ func (b *Cloud) opPlan(stopCtx, cancelCtx context.Context, op *backend.Operation
 		return nil, diags.Err()
 	}
 
-	return b.plan(stopCtx, cancelCtx, op, w)
+	// If the run errored, exit before checking whether to save a plan file
+	run, err := b.plan(stopCtx, cancelCtx, op, w)
+	if err != nil {
+		return run, err
+	}
+
+	// Maybe save plan file
```

Comment:
💅 Elaborate on the "maybe". What instances should a user expect the plan to be saved (not including the expected behavior when everything goes well). Or will this behavior be documented somewhere else?

Reply:
👍🏼done. (We save if you said -out.)